server/src/pages/shot/server.js

```diff
@@ -12,7 +12,7 @@ app.get("/:id/:domain", function(req, res) {
   const shotId = `${req.params.id}/${req.params.domain}`;
   Shot.get(req.backend, shotId).then((shot) => {
     const noSuchShot = !shot;
-    const nonOwnerAndBlocked = shot && shot.blockType !== "none" && req.deviceId !== shot.ownerId;
+    const nonOwnerAndBlocked = shot && shot.blockType !== "none" && !shot.isOwner(req.deviceId, req.accountId);
```

Comment:
We should include something about this in the QA plan (blocked shots and FxA login)